### PR TITLE
feat: navigation internationalization [CU-29pdxpx]

### PR DESCRIPTION
**Pull request by CodeVoyager**
## Note

This branch is based on `feat/typescript`

## Summary

- i18n capabilities for navigations
- new bootstrapping strategies
- TS typings updates

## Test Plan

- start application (expected result: everything stays the same)
- set default system language
- add other locales (like: `fr` or `pl`)
- add content types that use internationalisation
- enable i18n
  - new navigation versions are created
  - on navigation editing one can pick locale version
  - while editing locale version content types that are i18n and not at current locale should not be selectable
  - content without i18n should work as it used to
  - navigations should work as they are described in docs
  - graphQL schema should be aware of locale setting
- disable i18n
  - if navigations are not selected for pruning, they are still available
  - navigations are not grouped by language

### CODE REVIEW

File: server/common/constant.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export const DEFAULT_NAVIGATION_POPULATED_FIELDS = ["localizations"];
```
[the remark this answers is not in the document]
Comment (CodeVoyager) [reply server/common/constant.ts:1]: Sure, makes sense

File: admin/src/pages/View/components/NavigationHeader/index.js [hunk 2]
```diff
@@ -9,41 +9,86 @@ import { getTrad } from '../../../../translations';
...
+                    value={activeNavigation?.id}
+                    size="S"
+                  >
+                    {allLocaleVersions.map(({ id, localeCode }) => <Option key={id} value={id}>{localeCode}</Option>)}
+                  </Select> 
+                </GridItem>
+              : null
+            }
+            <GridItem col="3">
+              <Button
```
[the remark this answers is not in the document]
Comment (CodeVoyager) [reply admin/src/pages/View/components/NavigationHeader/index.js:81]: Yup, it's a bug. Will fix.

**Review by cyp3rius — APPROVED** [review]

File: server/graphql/queries/render-navigation-child.js [hunk 3]
```diff
@@ -10,7 +10,8 @@ module.exports = ({ strapi, nexus }) => {
     },
     resolve(obj, args) {
       const { id, childUIKey, type, menuOnly } = args;
-      return strapi.plugin('navigation').service('client').renderChildren(id, childUIKey, type, menuOnly);
+      const service = strapi.plugin('navigation').service('client')
```
[the remark this answers is not in the document]
Comment (CodeVoyager) [reply server/graphql/queries/render-navigation-child.js:13]: Will add generic type arguments on migration to TS.

File: admin/src/pages/SettingsPage/index.js [hunk 4]
```diff
@@ -275,10 +331,10 @@ const SettingsPage = () => {
                             hint={getMessage('pages.settings.form.allowedLevels.hint')}
                             onValueChange={(value) => setFieldValue('allowedLevels', value, false)}
                             value={values.allowedLevels}
-                            disabled={isRestartRequired}
+                            disabled={restartStatus.required}
                           />
                         </GridItem>
-                        <GridItem col={6} s={12} xs={12}>
+                        <GridItem col={4} s={12} xs={12}>
```
[the remark this answers is not in the document]
Comment (CodeVoyager) [reply admin/src/pages/SettingsPage/index.js:337]: I will provide alternative in the next PR if this issue is not critical

**Review by ltsNotMike — APPROVED** [review]

**Pull request merged**